### PR TITLE
fix 864 rewritten in react Sample Configuration Menu page

### LINKED ISSUE #864: Re- write Sample Configuration Menu page to the new React UI
https://oe3.openelis-global.org/api/OpenELIS-Global/SampleEntryConfigMenu

### PR DESCRIPTION
**Pull request by adityadeshlahre**
#### WIP

- [x] The PR title includes a brief description of the work done, including the Issue number if applicable.
- [x] The PR includes a video showing the changes for the work done.
- [x] The PR title follows conventional commit label standards.
- [x] The changes confirm to the OpenElis Global x3 Styleguide and design documentation.
- [x] The changes include tests or are validated by existing tests.
- [x] I have read and agree to the Contributing Guidelines of this project.

## Summary

rewritten in react Sample Configuration Menu page

og jsp component link : https://oe3.openelis-global.org/api/OpenELIS-Global/SampleEntryConfigMenu

### Video

sampleEditConfig

[Add relevant screenshots here if applicable]

## Related Issue

[Add a link to the related issue or mention it here if applicable]

## Other

[Add any additional information or notes here]

### CODE REVIEW

**Comment by mozzy11** [comment]
Thanks @adityadeshlahre for working on this.
Any progress or blocker ??

**Comment by adityadeshlahre** [comment]
image


Hey @mozzy11 I wasn't able to figure it out how to fetch the list of configuration properties and their boolean values.
Am on the way to creating restController for this but there are n number configuration properties and am confused on how to fetch there boolean values. [ on the basis of that boolValues on selection of row I will enable to disable the top buttons ]

Thank You :slightly_smiling_face:

**Comment by adityadeshlahre** [comment]
image


Hey @mozzy11 was able to get date from restController to frontend moving forward to fix the post part !

Previously issue was after building the restController I needed to fresh/recreate the openElis-WebApp container that fixed the issue relate to fetch call [ don't know the exact reason ]

**Comment by adityadeshlahre** [comment]
oec


### Current WorkFlow

File: frontend/src/components/admin/Admin.js [hunk 1]
```diff
@@ -84,6 +86,20 @@ function Admin() {
               defaultMessage={"Common Properties"}
             />
           </SideNavLink>
+          <SideNavMenu
+            title={intl.formatMessage({ id: "admin.react" })}
+            renderIcon={CloudSatelliteConfig}
+          >
+            <SideNavMenuItem href="#configurationManagement">
```
Comment (mozzy11) [inline frontend/src/components/admin/Admin.js:93]: Why did you have to add multiple Links to the sampe page . 
We only need one . 
it should be labeled `Order Entry Configuration` as on the old UI

File: frontend/src/components/admin/Admin.js [hunk 2]
```diff
@@ -124,6 +140,15 @@ function Admin() {
       <PathRoute path="#commonproperties">
         <CommonProperties />
       </PathRoute>
+      <PathRoute path="#configurationManagement">
+        <SampleEntryConfigMenu />
+      </PathRoute>
```
Comment (mozzy11) [inline frontend/src/components/admin/Admin.js:145]: why add multiple routes ??

File: frontend/src/components/admin/menu/SampleEntryConfigMenu.js [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+            <Section>
+              <Heading>
+                <FormattedMessage id="admin.orderEntryConfigMenu" />
+              </Heading>
+            </Section>
+            <br />
+            <br />
+            <Section>
+              <Form onSubmit={handleModify}>
+                <div style={{ margin: "1rem", display: "flex", gap: "1rem" }}>
```
Comment (mozzy11) [inline frontend/src/components/admin/menu/SampleEntryConfigMenu.js:169]: can you use the Carbon `Grid `and `Colum`elements for the Layout ??

File: frontend/src/components/admin/menu/SampleEntryConfigMenu.js [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+  const [modifyButton, setModifyButton] = useState(true);
+  const [selectedRowId, setSelectedRowId] = useState(null);
+  const [startingRecNo, setStartingRecNo] = useState(1);
+  const [sampleEntryConfigMenuList, setSampleEntryConfigMenuList] = useState(
+    [],
+  );
+  const [orderEntryConfigurationList, setOrderEntryConfigurationList] =
+    useState([]);
+
+  // async function displayStatus(res) {
```
Comment (mozzy11) [inline frontend/src/components/admin/menu/SampleEntryConfigMenu.js:58]: remove unused code

**Review by mozzy11 — CHANGES_REQUESTED** [review]
seecomments above

File: frontend/src/components/admin/menu/SampleEntryConfigMenu.js [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+  //     JSON.stringify(menuItem),
+  //     displayStatus,
+  //   );
+  // }
+
+  function handleModify(event) {
+    event.preventDefault();
+    let sampleEntryConfigEditRedirect =
+      config.serverBaseUrl +
+      `/SampleEntryConfig?ID=${selectedRowId}&startingRecNo=${startingRecNo}`;
```
Comment (mozzy11) [inline frontend/src/components/admin/menu/SampleEntryConfigMenu.js:92]: We need also to re-write the SampleEntryConfig config in react. 
You can just keep redirecting to the OLD UI for modifying the values

File: frontend/src/components/admin/Admin.js (hunk 1 above)
Comment (adityadeshlahre) [reply frontend/src/components/admin/Admin.js:93]: These changes are according to grouping of same configurations which I proposed to you in one of the community meetings Right now we don't have all the page so I added <OrderEntryConfiguration/> as page for now !

I will update it no worries :slightly_smiling_face:

File: frontend/src/components/admin/menu/SampleEntryConfigMenu.js (hunk 5 above)
Comment (adityadeshlahre) [reply frontend/src/components/admin/menu/SampleEntryConfigMenu.js:92]: alright mosses :slightly_smiling_face:

**Comment by adityadeshlahre** [comment]
#### UI updated WorkFlow of SampleConfigurationMenu/OrderEntryConfiguration

updateOEC

I think its ready to merge. If still any issue please let me know :slightly_smiling_face: 
Thank You

**Comment by adityadeshlahre** [comment]
> Thanks @adityadeshlahre . I can see youre still redirecting to the old UI for the actual modification

I think for most of the value modification page is same. So I think we should create a component which will display the props received from main OrderEnteryConfig page by selecting the name[which is displayed in name column in image above ] & Discription and there values which need to be editted!

But later on I realise that the description of all the configs are different and didn't getting in JSON response !
In JSON response the description which need to show in sampleConfigPage is not available.

am still finding solution for it !

will update the PR accordingly :slightly_smiling_face: 
Thank You

**Comment by adityadeshlahre** [comment]
Sample Edit Config UI 

created this separate component to receive name and description

image

**Comment by mozzy11** [comment]
Hello @adityadeshlahre .
Thanks for working on this . This PR https://github.com/I-TECH-UW/OpenELIS-Global-2/pull/1022 implements a reussable component for all the configuration pages. 
Can you pull in those changes and try to see if thres any pending improvements needed ??

**Comment by adityadeshlahre** [comment]
This PR is not required. 
Since the same thing is implemented in #1022

**Pull request closed without merging**